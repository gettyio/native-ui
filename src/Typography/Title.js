import styled from 'styled-components/native'

const Body = styled.Text`
  font-size: 37px;
  font-weight: normal;
  line-height: 55px;
  letter-spacing: 0.176;
`

export default Body
