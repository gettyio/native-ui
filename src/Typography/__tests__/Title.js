/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Title from '../Title'

describe('Title component', () => {
  it('should render without issue', () => {
    const component = shallow(<Title>Title text.</Title>)

    expect(component.length).toBe(1)
    expect(toJson(component)).toMatchSnapshot()
  })

  it('should have default props', () => {
    const component = shallow(<Title>Title text.</Title>)

    expect(toJson(component)).toMatchSnapshot()
    expect(toJson(component)).toHaveStyleRule('font-weight', 'normal')
    expect(toJson(component)).toHaveStyleRule('font-size', 37)
    expect(toJson(component)).toHaveStyleRule('line-height', 55)
    expect(toJson(component)).toHaveStyleRule('letter-spacing', 0.176)
  })
})
