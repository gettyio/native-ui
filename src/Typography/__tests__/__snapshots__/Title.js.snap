// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component should have default props 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "fontSize": 37,
        "fontWeight": "normal",
        "letterSpacing": 0.176,
        "lineHeight": 55,
      },
      undefined,
    ]
  }
>
  Title text.
</Text>
`;

exports[`Title component should render without issue 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "fontSize": 37,
        "fontWeight": "normal",
        "letterSpacing": 0.176,
        "lineHeight": 55,
      },
      undefined,
    ]
  }
>
  Title text.
</Text>
`;
